Neos.Media/Classes/Domain/Model/ImportedAssetManager.php

```diff
@@ -0,0 +1,102 @@
+<?php
+namespace Neos\Media\Domain\Model;
+
+/*
+ * This file is part of the Neos.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+  *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations\Inject;
+use Neos\Flow\Log\SystemLoggerInterface;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Media\Domain\Model\AssetSource\AssetSourceAwareInterface;
+use Neos\Media\Domain\Repository\ImportedAssetRepository;
+
+class ImportedAssetManager
+{
+    /**
+     * @Inject()
+     * @var  PersistenceManagerInterface
+     */
+    protected $persistenceManager;
+
+    /**
+     * @Inject()
+     * @var ImportedAssetRepository
+     */
+    protected $importedAssetRepository;
+
+    /**
+     * @Inject()
+     * @var SystemLoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @param ImportedAsset $importedAsset
+     * @throws \Neos\Flow\Persistence\Exception\IllegalObjectTypeException
+     */
+    public function registerImportedAsset(ImportedAsset $importedAsset)
+    {
+        $this->importedAssetRepository->add($importedAsset);
+        $this->logger->log(sprintf('Asset imported: %s / %s as %s', $importedAsset->getAssetSourceIdentifier(), $importedAsset->getRemoteAssetIdentifier(), $importedAsset->getLocalAssetIdentifier()), LOG_DEBUG);
+    }
+
+    /**
+     * Register that an asset was created.
+     *
+     * Wired via signal-slot with AssetService::assetCreated – see Package.php
+     *
+     * @param AssetInterface $asset
+     * @throws \Neos\Flow\Persistence\Exception\IllegalObjectTypeException
+     */
+    public function registerCreatedAsset(AssetInterface $asset)
+    {
+        if (!$asset instanceof AssetVariantInterface || !$asset instanceof AssetSourceAwareInterface) {
+            return;
+        }
+
+        $variantAsset = $asset;
```

Comment:
I think the renaming is unneeded here, or if its for making things clearer, it should be used in the next line